mrphowl/javascript-track
# JS-121: Using SQL ORM in Node.JS

# Related Ticket

https://codeandtheory.atlassian.net/browse/JS-121

# Description

Workshop project files.

# Validation

1. Install dependencies
    ```sh
    > npm install
    ```
2. Launch the application
    ```sh
    > npm start
    ```
3. Open http://localhost:3000 in a browser
4. User should be able to do the following:
    - View the article listing
        <img width="500" alt="image" src="https://user-images.githubusercontent.com/8042132/184632557-9855df7f-08bf-4faa-b849-a4162d3656d8.png">
    - Create an article
        - Click the **Create Article +** button to go to the new article form
            <img width="100" alt="image" src="https://user-images.githubusercontent.com/8042132/184632135-e080178d-ace1-48b9-90c0-903dfa482b97.png">
        - Fill in the details
        - Click the **Submit** button
    - View the article content
        <img width="500" alt="image" src="https://user-images.githubusercontent.com/8042132/184632774-143c8c5d-549d-411d-883f-0d2b5033af4e.png">
    - Edit the details of an article
        <img width="500" alt="image" src="https://user-images.githubusercontent.com/8042132/184633481-f7f5eed7-3022-40af-ab45-c6384b14b762.png">
    - Delete an article
        <img width="300" alt="image" src="https://user-images.githubusercontent.com/8042132/184633599-80bfc391-19bf-45d6-8c0c-3a3143d4398e.png">

5. Creation and editing of an article should have field validation
    - Title cannot be empty
    - Title cannot exceed 30 characters
    - Author cannot be empty
    <br>
    <img width="600" alt="image" src="https://user-images.githubusercontent.com/8042132/184631733-38e9ba4a-1a2b-4279-adc1-49c68b83d903.png">

    <img width="470" alt="image" src="https://user-images.githubusercontent.com/8042132/184631975-aba96ef9-c866-4936-b878-fe4d86d57f29.png">


6. The application should show **Not found** for non-existing article IDs.

    <img width="300" alt="image" src="https://user-images.githubusercontent.com/8042132/184631607-b47ff784-7231-4879-832f-68e3e33d1478.png">


# Risk

None
